File: tgprintf.m
function ret = tgprintf(varargin)
% TGPRINTF send a message to a Telegram bot
%
% Use tgprintf() in the same way as sprintf()
% Example: tgprintf('Hello, World!');
%          tgprintf('%d + %d = %d',1,2,1+2);
% 
% Define token and chat_id before use, 
% which are the authorization token of the target Telegram bot 
% and the identifier or username of the target chat
%
% Please refer the following post 
% "Creating a Telegram bot for personal notifications"
% https://www.forsomedefinition.com/automation/creating-telegram-bot-notifications/
% 
% Seongsik Park
% seongsikpark@postech.ac.kr

% default token and chat_id
token = DEFAULT_TOKEN_HERE;
chat_id = DEFAULT_CHAT_ID_HERE;

str = sprintf(varargin{:});

% print to MATLAB command window
fprintf(str);

% convert MATLAB string to url query string
sendstr = urlencode(str);
sendstr = ['https://api.telegram.org/bot',token,...
           '/sendMessage?chat_id=',chat_id,...
           '&text=',sendstr];

% send a message   
ret = webread(sendstr); 
assert(ret.ok);

% append human readable datetime to results [Set TimeZone value to desired time zone]
ret.result.datetime=datetime(ret.result.date,'ConvertFrom','posixtime','TimeZone','Asia/Seoul');
end

File: tgprintf.cpp
#include <string>
#include <iostream>

using namespace std;

wstring s2ws(const string& s)
{
	int len;
	int slength = (int)s.length() + 1;
	len = MultiByteToWideChar(CP_ACP, 0, s.c_str(), slength, 0, 0);
	wchar_t* buf = new wchar_t[len];
	MultiByteToWideChar(CP_ACP, 0, s.c_str(), slength, buf, len);
	std::wstring r(buf);
	delete[] buf;
	return r;
}

void tgprintf(string hello) {
	string chat_id = "59233348";
	string token = "430183280:AAEoNtMwQQFxMhE7VP9hwQnHWI-RAtyD9No";
	string str = "https://api.telegram.org/bot" + token
		+ "/sendMessage?chat_id=" + chat_id + "&text=" + hello;
	
	wstring send = s2ws(str);
	printf("%s", str.c_str());
	ShellExecute(0, 0, send.c_str(), 0, SW_HIDE);
}
